### DISCUSSION TITLE
Sign with Azure Key Vault

### DISCUSSION THREAD

**Discussion by naviprobeha** [microsoft/AL-Go#1335] [Q&A] [answered]
I'm doing the migration of my signing certificate to the Azure Key Vault. But when running the CI/CD pipeline, it complains that the runner cannot find "dotnet"

Error: Unexpected error when running action. Error Message: The term 'dotnet' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again., StackTrace: at Install-SigningTool, C:\actions-runner\_work\_actions\microsoft\AL-Go-Actions\v6.1\Sign\Sign.psm1:

When opening a command prompt on the runner, the dotnet command executes fine. In what context is the script trying to run the command?

Regards

**Comment by freddydk** [comment]
Could you include the full log of the sign job?

**Comment by aholstrup1** [comment]
If this is a self-hosted runner please make sure you've followed all the steps here: 
https://github.com/microsoft/AL-Go/blob/main/Scenarios/SelfHostedGitHubRunner.md#create-your-self-hosted-runner-manually

e.g. make sure .NET is installed on the agent

**Comment by freddydk** [comment]
Latest AL-Go requires dotnet 6.1 btw.

**Comment by naviprobeha** [comment]
Thanks!!

Dotnet was installed, but I followed the procedures from the attached instructions all over.

Now I get this error message:

Module Az.Accounts is available in version 4.0.0
Importing module Az.Accounts (minimum version 2.15.1)
Module Az.KeyVault is available in version 6.3.0
Importing module Az.KeyVault (minimum version 5.2.0)
Connecting to Azure using clientId and clientSecret.
Selecting subscription ***
Successfully connected to Azure
Signing files
  Installing signing tool version 0.9.1-beta.24469.1 in C:\Users\github_service\AppData\Local\Temp\SigningTool-614325231
  No NuGet sources are defined or enabled
  Applying settings from C:\actions-runner\_work\AnywareRetail\AnywareRetail\.github\AL-Go-Settings.json
  Applying settings from C:\actions-runner\_work\AnywareRetail\AnywareRetail\.AL-Go\settings.json
  No settings found in C:\actions-runner\_work\AnywareRetail\AnywareRetail\.github\CICD.settings.json
  No settings found in C:\actions-runner\_work\AnywareRetail\AnywareRetail\.AL-Go\CICD.settings.json
  No settings found in C:\actions-runner\_work\AnywareRetail\AnywareRetail\.AL-Go\naviprobeha.settings.json
  Enabling Microsoft telemetry...
  Error: Unexpected error when running action. Error Message: Nuget source https://api.nuget.org/v3/index.json is not added. Please add the source using 'dotnet nuget add source https://api.nuget.org/v3/index.json' or add another source with nuget.org as an upstream source., StackTrace: at AssertNugetSourceIsAdded, C:\actions-runner\_work\_actions\microsoft\AL-Go-Actions\v6.1\Sign\Sign.psm1: line 9 <- at Install-SigningTool, C:\actions-runner\_work\_actions\microsoft\AL-Go-Actions\v6.1\Sign\Sign.psm1: line 37 <- at Invoke-SigningTool, C:\actions-runner\_work\_actions\microsoft\AL-Go-Actions\v6.1\Sign\Sign.psm1: line 111 <- at <ScriptBlock>, C:\actions-runner\_work\_actions\microsoft\AL-Go-Actions\v6.1\Sign\Sign.ps1: line 69 <- at <ScriptBlock>, C:\actions-runner\_work\_temp\8d0c4b0b-0379-4[54](https://github.com/NaviProAB/AnywareRetail/actions/runs/11931590917/job/33534082674#step:9:57)5-96d2-7be5d033f79b.ps1: line 3 <- at <ScriptBlock>, C:\actions-runner\_work\_actions\microsoft\AL-Go-Actions\v6.1\Invoke-AlGoAction.ps1: line 17 <- at <ScriptBlock>, C:\actions-runner\_work\_temp\8d0c4b0b-0379-4545-96d2-7be5d033f79b.ps1: line 2 <- at <ScriptBlock>, <No file>: line 1
  Error: Process completed with exit code 1.

However, the nuget source is added:

C:\Windows\system32>dotnet nuget list source
Registered Sources:
  1.  nuget.org [Enabled]
      https://api.nuget.org/v3/index.json

C:\Windows\system32>


Any ideas?

**Reply by aholstrup1** [reply, marked as answer]
I'm guessing the nuget source is added for a different user than the one used by GitHub. 

See similar issue here: 
https://github.com/microsoft/AL-Go/issues/1252#issuecomment-2408915315

**Comment by naviprobeha** [comment]
That was the issue. Thanks!!
